Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -65,9 +78,27 @@ namespace ROS2
 
         if (m_publishTransform)
         {
+            AZ_TracePrintf("ROS2FrameComponent", "Setting up %s", GetFrameID().data());
+
+            // The frame is static when is not a top entity and one of the following is true:
+            //    - if the entity has at least one Fixed Joint
+            //    - if the entity has no joints at all.
+            if (IsTopLevel())
+            {
+                m_isDynamic = true;
+            }
+            else
+            {
+                bool hasJoints = Internal::CheckIfEntityHasComponentOfType(
+                    m_entity, AZ::Uuid("{B01FD1D2-1D91-438D-874A-BF5EB7E919A8}")); // Physx::JointComponent;
+                bool hasFixedJoints = Internal::CheckIfEntityHasComponentOfType(
```

Comment:
nit: This is super minor but we general try and mark everything that doesn't change `const`, so these could have `const` in front of them